libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1704,14 +1704,24 @@ void GCS_MAVLINK::handle_statustext(mavlink_message_t *msg)
     if (df == nullptr) {
         return;
     }
-    // ignore any statustext messages not from our GCS:
-    if (msg->sysid != sysid_my_gcs()) {
-        return;
-    }
+
     mavlink_statustext_t packet;
     mavlink_msg_statustext_decode(msg, &packet);
-    char text[MAVLINK_MSG_STATUSTEXT_FIELD_TEXT_LEN+1+4] = { 'G','C','S',':'};
-    memcpy(&text[4], packet.text, MAVLINK_MSG_STATUSTEXT_FIELD_TEXT_LEN);
+    const uint8_t max_prefix_len = 20;
+    const uint8_t text_len = MAVLINK_MSG_STATUSTEXT_FIELD_TEXT_LEN+1+max_prefix_len;
+    char text[text_len] = { 'G','C','S',':'};
+    uint8_t offset = strlen(text);
+
+    if (msg->sysid != sysid_my_gcs()) {
+        offset = hal.util->snprintf(text,
```

Comment:
This was this on purpose to overwrite the *GCS:* part, right? I think keeping it would be nicer, by making clear it came from a remote system. Although I think you've done it to differentiate "my GCS" from other sources (that might not be a GCS), having the *SRC* part only when it isn't from "my GCS" already makes it clear.